### DISCUSSION TITLE
how to increase window size on up/down/left/right?

### DISCUSSION THREAD

**Discussion by j1g5awi** [hyprwm/Hyprland#7591] [Q&A] [answered]
if I have a float window in center of screen, and I want to use Mod+h/j/k/l to increase its size, and Mod_Shift+h/j/k/l to decrease its size on different direction, how can i config key map?

the dispatcher `resizeactive` looks like only increase size on two direction, and others decrease.

**Comment by awarewen** [comment]
- try this, works fine

```
bind = $MAIN_MOD, R, submap, resize                                     # <MAIN+R ( )>
submap=resize
  binde = , H, resizeactive, -20  0                     # <H>
  binde = , J, resizeactive,  0   20                    # <J>
  binde = , K, resizeactive,  0  -20                    # <K>
  binde = , L, resizeactive,  20  0                     # <L>
  binde = , Left, resizeactive, -10  0                     # <Left>
  binde = , Down, resizeactive,  0   10                    # <Down>
  binde = , Up,   resizeactive,  0  -10                    # <Up>
  binde = , Right, resizeactive,  10  0                     # <Right>
# Quit : 退出一级 submap
bind  = , R,      submap, reset                                         # <R>
bindt = , escape, submap, reset                                         # <ESC>
submap = reset
```

**Reply by j1g5awi** [reply]
first of all, you misunderstand my question, maybe I didn't express it clearly. I mean **anytime I use h/j/k/l, it always increase windows size; anytime I press shift+hjkl, it always decrease window size.**. now `resizeactive` can only make h increase in right, and l decrease in right.

to prevent misunderstanding again, here is the Chinese translation of above：你误解了我的问题，我想要的是 **按 h/j/k/l 的時候都是增大窗口，按 shift+hjkl 的時候都是减小窗口**。目前 `resizeactive` 只能做到在两个方向（上右、右下、下左、左上）上增减窗口大小——负数是减小，而非在另一方向上增大。

second, you should test you config before you answer a question. **you answer is wrong, resizeactive is not a shell command to exec, it's a hyprland dispatcher.**

true example to use `resizeactive` like below:

```hypr
bind=$mainMod,R,submap,resize
submap=resize
binde=,l,resizeactive,10 0
binde=,h,resizeactive,-10 0
binde=,k,resizeactive,0 -10
binde=,j,resizeactive,0 10
bind=,escape,submap,reset
submap=reset
```

**Reply by erwin** [reply]
I have the same question.
I would like to do exactly the same thing as you.
Be able to grow left/shrink left or grow right/shrink right (potentially same for up/down)

```                        
      ┌─────────────┐      
   ◄──┼──►       ◄──┼──►   
      │             │      
      │             │      
      │             │      
      │             │      
   ◄──┼──►       ◄──┼──►   
      └─────────────┘      
```                           

我想要的是 按 h/j/k/l 的時候都是增大窗口，按 shift+hjkl 的時候都是减小窗口。目前 resizeactive 只能做到在两个方向（上右、右下、下左、左上）上增减窗口大小——负数是减小，而非在另一方向上增大

From what I can see, it's currently not possible to resize the windows in this way in hyprland.

From what I've been able to figure out so far, I only see how to grow/shrink the right edge of the window, but not the left edge

**Comment by ghost** [comment, marked as answer]
@erwin, and  @j1g5awi (if you are still following this)  In order to be able to grow or shrink all sides you need to move the window simultaneously for growing/shrinking it up or left:
```
# grow

binde=$mainMod,h,resizeactive,20 0
binde=$mainMod,h,moveactive,-20 0
binde=$mainMod,l,resizeactive,20 0

binde=$mainMod,k,resizeactive,0 20
binde=$mainMod,k,moveactive,0 -20
binde=$mainMod,j,resizeactive,0 20

# shrink

binde=$mainMod SHIFT,h,resizeactive,-20 0
binde=$mainMod SHIFT,h,moveactive,20 0
binde=$mainMod SHIFT,l,resizeactive,-20 0

binde=$mainMod SHIFT,k,resizeactive,0 -20
binde=$mainMod SHIFT,k,moveactive,0 20
binde=$mainMod SHIFT,j,resizeactive,0 -20
```

Because the windows are anchored in the top left, you can resize right and bottom simply with the `resize cmd`. 

For the other ones you need to have a counteracting, moving force bound to the same key. You are still growing the window on the wrong side but because you move it the same amount to the preferred side it is visually indistinguishable.

**Reply by j1g5awi** [reply]
I'm glad to find answer after 2 years, unfortunally I have used to default behavior of riszeactivef. 

actually, **resizeactive 20,0** will grow window's width in 20 pixel, 10 in left, 10 in right. so config most close to my purpose is:

```
binde=$mainMod,h,resizeactive,20 0
binde=$mainMod,h,moveactive,-10 0
binde=$mainMod,l,resizeactive,20 0
binde=$mainMod,l,moveactive,10 0
```

**Reply by ghost** [reply]
Awesome to hear that the moveactive binding worked for you!

**Comment by erwin** [comment]
@j1g5awi 

Thank you for the support and help!

That makes sense. I guess I'll implement a "SUPER+R" sub-menu for moves and resizes floating windows with vi keys... 

Sure wish this logic could also be applied to regular tiled windows, especially makes sense for anyone else using `layout = scrolling`

The current way, you often have to switch to the adjacent window and shrink or grow it to get another window to the size you want

**Reply by ghost** [reply]
I'm not sure if this is what you want but when I am in tiling mode (never tried scrolling) this seems to work:
```
binde=$mainMod,h,resizeactive,-20 0
binde=$mainMod,l,resizeactive,20 0
binde=$mainMod,k,resizeactive,0 20
binde=$mainMod,j,resizeactive,0 -20
```
You may have to remove this line in your config first if you haven't already:
```
bind = $mainMod, J, togglesplit, # dwindle
```
And of course find a way to define the other ones for the floating windows somewhere else.
Hope this helps.
